services/advertise.js

```diff
@@ -124,9 +168,18 @@ module.exports = (args, cbk) => {
       nodes: [
         'getGraph',
         'getIdentity',
+        'getTags',
         'message',
-        ({getGraph, getIdentity, message}, cbk) =>
+        ({getGraph, getIdentity, getTags, message}, cbk) =>
       {
+        // Exit early if tags are present
+        if (!!getTags) {
+          const nodes = getTags.filter(n => n !== getIdentity.public_key)
+            .map(n => ({public_key: n}));
+
+          return cbk(null, nodes);
```

Comment:
Should this use `getNodesGraph` instead? I think if it doesn't then the filters won't work